### PR TITLE
Derive APIs Explorer link from service config name

### PR DESCRIPTION
**Pull request by geigerj**
Previously, it was assumed this was the same as the artman "short
name" for the API, which is not necessarily true.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java [hunk 1]
```diff
@@ -27,6 +27,14 @@
       String template,
       String outputPath,
       TargetLanguage language) {
+    // Note that internally, this is overridable using a "discovery" section in the service config,
```
Comment (ethanbao) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: discovery is not part of the public service config, so let's not mention it here.

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java [hunk 2]
```diff
@@ -27,6 +27,14 @@
...
+    // which is not available externally. See:
+    //   https://github.com/googleapis/toolkit/issues/933
+    String discoveryApiName = model.getServiceConfig().getName();
```
Comment (ethanbao) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: This is actually service name. And I would suggest naming the first segment as apiName without mentioning discovery.

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java [hunk 3]
```diff
@@ -47,6 +55,7 @@
         .homepage(packageConfig.homepage())
         .licenseName(packageConfig.licenseName())
         .fullName(model.getServiceConfig().getTitle())
+        .discoveryApiName(discoveryApiName)
```
Comment (ethanbao) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: need to replace dash with underscore.

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java (hunk 2 above)
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: The name `apiName` is confusing in the context of GAPIC given that we already have `shortName` and `fullName` to refer to the API's name. I want to clarify that this value should be used where the name of the discovery doc is needed.

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java (hunk 1 above)
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: Done

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java (hunk 3 above)
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: Done

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java (hunk 2 above)
Comment (ethanbao) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: Ack

**Comment by ethanbao** [comment]
LGTM

**Pull request merged**